scheds/rust/scx_bpfland/src/bpf/main.bpf.c

```diff
@@ -660,21 +659,45 @@ s32 BPF_STRUCT_OPS(bpfland_select_cpu, struct task_struct *p,
 static void kick_idle_cpu(const struct task_struct *p, const struct task_ctx *tctx)
 {
 	const struct cpumask *idle_cpumask;
+	const struct cpumask *mask;
 	s32 cpu;
 
+	idle_cpumask = scx_bpf_get_idle_cpumask();
+
 	/*
 	 * Look for any idle CPU usable by the task that can immediately
 	 * execute the task.
 	 *
 	 * Note that we do not want to mark the CPU as busy, since we don't
 	 * know at this stage if we will actually dispatch any task on it.
 	 */
-	idle_cpumask = scx_bpf_get_idle_cpumask();
-	cpu = bpf_cpumask_any_and_distribute(p->cpus_ptr, idle_cpumask);
-	scx_bpf_put_cpumask(idle_cpumask);
+	mask = cast_mask(tctx->l2_cpumask);
```

Comment:
The one thing I've thought of for the the selection is it is better to do something like this in the SMT case:
L2 -> check idle SMT -> check idle CPU
L3 -> check idle SMT -> check idle CPU

On partially loaded SMT systems I wonder if it's better to try to get better SMT isolation vs L2 locality. I haven't done any testing, but curious if you've thought of that. The downside of checking if there's an idle SMT at the L2/L3 level first is a bunch of extra code though.

Reply:
That is probably a good idea, I'll do some tests!